protocol-units/da-sequencer/node/src/server.rs

```diff
@@ -59,7 +65,116 @@ impl DaSequencerNodeService for DaSequencerNode {
+		let (produced_tx, mut produced_rx) = mpsc::unbounded_channel();
+		let (curent_height_tx, curent_height_rx) = oneshot::channel();
+		if let Err(err) = self
+			.request_tx
+			.send(GrpcRequests::StartBlockStream(produced_tx, curent_height_tx))
+			.await
+		{
+			tracing::warn!("Internal grpc request channel closed, can't stream blocks:{err}");
+			return Err(tonic::Status::internal("Internal error. Retry later"));
+		}
+
+		let mut current_produced_height = match curent_height_rx.await {
+			Ok(h) => h,
+			Err(err) => {
+				tracing::warn!("Get start stream block oneshot channel closed: {err}");
+				return Err(tonic::Status::internal("Internal error. Retry later"));
+			}
+		};
+		let mut current_block_height = request.into_inner().height;
+		//Genesis block can't be retrieved.
+		if current_block_height == 0 {
+			current_block_height = 1;
+		}
+		let request_tx = self.request_tx.clone();
+		let output = async_stream::try_stream! {
+			loop {
+				let response_content = if current_block_height <= current_produced_height.0 {
+					//get all block until the current produced height
+					let (get_height_tx, get_height_rx) = oneshot::channel();
+					if let Err(err) = request_tx.send(GrpcRequests::GetBlockHeight(
+						current_block_height.into(),
+						get_height_tx,
+					)).await {
+						tracing::warn!("Stream block get block at height oneshot channel closed, can't stream blocks:{err}");
+						return;
+					}
+					let block = match get_height_rx.await {
+						Ok(b) => b,
+						Err(err) => {
+							tracing::warn!("Stream block, stream block oneshot channel closed: {err}");
+							return;
+						}
+					};
+					current_block_height +=1;
+
+					let blockv1 = match block.as_ref().map(|block| block.try_into()) {
+						None => continue,
+						Some(Ok(b)) => b,
+						Some(Err(err)) => {
+							tracing::warn!("Stream block block serialization failed :{err}");
+							return;
+
+						}
+					};
+					BlobResponse { blob_type: Some(BlobType::Blockv1(blockv1)) }
+
+				} else {
+					//send block in produced channel
+					let received_content = match produced_rx.recv().await {
+						Some(b) => b,
+						None => {
+							tracing::warn!("Stream block produced block channel closed.");
+							return;
+						}
+
+					};
+
+					match received_content {
+						None => {
+							// send heartbeat.
+							BlobResponse { blob_type: Some(BlobType::Heartbeat(true)) }
+						}
+
+						Some(new_block) => {
+							// send new produced block.
+							if current_block_height + 1 < new_block.height.0 {
+								// we miss a block request it + the one we get.
+								current_produced_height = new_block.height;
+								continue;
+							}
+							current_block_height = new_block.height.0;
+
+							let blockv1 = match new_block.try_into() {
+								Ok(b) => b,
+								Err(err) => {
+									tracing::warn!("Stream block block serialization failed :{err}");
```

Comment:
```suggestion
									tracing::warn!("Stream block: block serialization failed :{err}");
```
Is that what you mean? A colon to distinguish where the error is coming from? Let's be more careful with these warn messages and the format we choose.